### PR TITLE
IntentSlotClassificationModel exportable

### PR DESCRIPTION
**Pull request by drnikolaev**
Signed-off-by: Sergei Nikolaev <snikolaev@nvidia.com>

### CODE REVIEW

**Review by tkornuta-nvidia — APPROVED** [review]
The PR looks good, one thing that makes me wonder is all those train/test data.
a) do we need all those samples?
b) I know that this is not binary, but do we want to keep it in here? 
Note that along with 0.11 we have moved all test data to files stored/distributed along with release:
https://github.com/NVIDIA/NeMo/releases/tag/v0.11.0
along with methods added to tests that download them from that URL.

Maybe we should do the same for 1.0, "update" the existing test files to and those files to release? @okuchaiev @blisc @titu1994 @ericharper ?

**Comment by drnikolaev** [comment]
Thanks Tom,
at this moment it's unavoidable, but let's talk to Vlad - the owner

**Comment by vladgets** [comment]
What are the questions to me?

**Comment by drnikolaev** [comment]
How can we instantiate the model without providing so much data?

**Comment by vladgets** [comment]
To make inference with a trained model basically we need only Dictionary of Intent and Slots, to know how to interpret the result numbers. They are 2 small text files stored in the data directory that is used for training. I can also add them inside of Config files, if it is better. For testing of training we need some training data as for any other model.

**Comment by drnikolaev** [comment]
No, just to build a graph and export it to ONNX.

File: nemo/core/classes/common.py [hunk 1]
```diff
@@ -356,6 +356,7 @@ def from_pretrained(
         model_name: str,
         refresh_cache: bool = False,
         override_config_path: Optional[str] = None,
+        override_data_dir: Optional[str] = None,
```
Comment (okuchaiev) [inline nemo/core/classes/common.py:359]: why this parameter was changed. Can't we control this via setup_training/validation_data methods?

Comment (okuchaiev) [reply nemo/core/classes/common.py:359]: Also, data_dir is in config file which can be overridden right now anyways.

**Review by okuchaiev — CHANGES_REQUESTED** [review]
can you please package all "dummy_data" as an artifact so that it gets downloaded with the rest of test data. (ping me offline for how-to).

File: nemo/core/classes/modelPT.py [hunk 2]
```diff
@@ -242,6 +243,8 @@ def restore_from(
                     # If override is top level config, extract just `model` from it
                     if 'model' in conf:
                         conf = conf.model
+                if override_data_dir is not None:
```
Comment (okuchaiev) [inline nemo/core/classes/modelPT.py:246]: @vladgets isn't data_dir part of config anyways?

Comment (vladgets) [reply nemo/core/classes/modelPT.py:246]: In Default config it is '???'. User should define it to refer to his dataset to train the model. We possibly had a strange issue with the model that I shared to Nvidia cloud is that I trained using my local data directory and it was automatically saved as a part of config in a .nemo file. Then when you download this model it refer to my local directory. So to use the pretrained model a user need redefine this parameter to his own data directory.

Comment (drnikolaev) [reply nemo/core/classes/modelPT.py]: Sorry about confusion, it's reverted now

File: nemo/core/classes/common.py (hunk 1 above)
Comment (drnikolaev) [reply nemo/core/classes/common.py]: yes, just mixed with other use case

**Review by okuchaiev — APPROVED** [review]

**Pull request merged**